lib/blueprinter/optimizer.rb

```diff
@@ -0,0 +1,36 @@
+module Blueprinter
+  class Optimizer
+    def initialize(object)
+      @object = object
+    end
+
+    def optimize!(fields)
+      return object unless active_record_relation?
```

Comment:
In addition to my factory comment, this guard could be checked before initialization.

Reply:
Good point. Though this will not be a problem after this class becomes a factory since we wont be initializing a new object.